pkg/api/validate/dynamic/dynamic.go

```diff
@@ -388,6 +392,29 @@ func (dv *dynamic) ValidateSubnets(ctx context.Context, oc *api.OpenShiftCluster
 	return nil
 }
 
+func (dv *dynamic) ValidateWorkerSku(ctx context.Context, location string, subscriptionID string, oc *api.OpenShiftCluster) error {
+	msiAuthorizer, err := dv.env.NewMSIAuthorizer(env.MSIContextRP, dv.env.Environment().ResourceManagerEndpoint)
+	if err != nil {
+		return err
+	}
+
+	resourceSkusClient := compute.NewResourceSkusClient(dv.env.Environment(), subscriptionID, msiAuthorizer)
```

Comment:
Any reason why this can't go into `NewValidator`? Similar to other clients? `NewMSIAuthorizer` should probably even be called outside: in the calling function.

There is no test coverage for `ValidateWorkerSku`. I think the fact that we can `New*` function is the reason for this as well.